pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -483,88 +512,167 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
+						resourceExport = &kdmpapi.ResourceExport{}
+						// Adding required label for debugging
+						labels := make(map[string]string)
+						labels[utils.ApplicationRestoreCRNameKey] = utils.GetValidLabel(restore.Name)
+						labels[utils.ApplicationRestoreCRUIDKey] = utils.GetValidLabel(utils.GetShortUID(string(restore.UID)))
+						// If restore from px-backup, update the restore object details in the label
+						if val, ok := backup.Annotations[utils.PxbackupAnnotationCreateByKey]; ok {
+							if val == utils.PxbackupAnnotationCreateByValue {
+								labels[utils.RestoreObjectNameKey] = utils.GetValidLabel(backup.Annotations[utils.PxbackupObjectNameKey])
+								labels[utils.RestoreObjectUIDKey] = utils.GetValidLabel(backup.Annotations[utils.PxbackupObjectUIDKey])
+							}
+						}
+						resourceExport.Labels = labels
+						resourceExport.Annotations = make(map[string]string)
+						resourceExport.Annotations[utils.SkipResourceAnnotation] = "true"
+						resourceExport.Name = crName
+						resourceExport.Namespace = restore.Namespace
+						resourceExport.Spec.Type = kdmpapi.ResourceExportBackup
+						// TODO: In the restore path we need to change source and destination ref as it is confusing now
+						// Usually dest means where it's backed up or restore to
+						source := &kdmpapi.ResourceExportObjectReference{
+							APIVersion: restore.APIVersion,
+							Kind:       restore.Kind,
+							Namespace:  restore.Namespace,
+							Name:       restore.Name,
+						}
+						backupLocation, err := storkops.Instance().GetBackupLocation(backup.Spec.BackupLocation, backup.Namespace)
+						if err != nil {
+							return fmt.Errorf("error getting backup location path: %v", err)
+						}
+						destination := &kdmpapi.ResourceExportObjectReference{
+							// TODO: GetBackupLocation is not returning APIVersion and kind.
+							// Hardcoding for now.
+							// APIVersion: backupLocation.APIVersion,
+							// Kind:       backupLocation.Kind,
+							APIVersion: "stork.libopenstorage.org/v1alpha1",
+							Kind:       "BackupLocation",
+							Namespace:  backupLocation.Namespace,
+							Name:       backupLocation.Name,
+						}
+						resourceExport.Spec.Source = *source
+						resourceExport.Spec.Destination = *destination
+						_, err = kdmpShedOps.Instance().CreateResourceExport(resourceExport)
+						if err != nil {
+							logrus.Errorf("failed to create DataExport CR: %v", err)
+							return err
+						}
+						return nil
+					}
+					return nil
+
+				} else {
+					backupVolInfos := convertResourceVolInfoAppVolInfoTo(resourceExport.VolumesInfo)
+					restoreVolumeInfos, sErr = driver.StartRestore(restore, backupVolInfos, nil)
+				}
+			}
+			if sErr != nil {
+				logrus.Infof("%s sErr: %v", funct, sErr)
 				message := fmt.Sprintf("Error starting Application Restore for volumes: %v", err)
 				log.ApplicationRestoreLog(restore).Errorf(message)
 				if _, ok := err.(*volume.ErrStorageProviderBusy); ok {
```

Comment:
Here also err -> sErr rt? looks like err to sErr need to take care in the below block of code